shapes/main.cpp

```diff
@@ -1,80 +1,68 @@
-#include <iostream>
-#include <vector>
 #include <algorithm>
-#include <string>
+#include <functional>
+#include <iostream>
+#include <map>
 #include <memory>
-#include "Shape.hpp"
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include "Circle.hpp"
 #include "Rectangle.hpp"
+#include "Shape.hpp"
 #include "Square.hpp"
-#include "Circle.hpp"
 
 using namespace std;
 
 using Collection = vector<shared_ptr<Shape>>;
 
-bool sortByArea(shared_ptr<Shape> first, shared_ptr<Shape> second)
-{
-    if(first == nullptr || second == nullptr)
+auto sortByArea = [](shared_ptr<Shape> first, shared_ptr<Shape> second) {    if (first == nullptr || second == nullptr)
         return false;
-    return (first->getArea() < second->getArea());
-}
+    return (first->getArea() < second->getArea()); };
 
-bool perimeterBiggerThan20(shared_ptr<Shape> s)
-{
-    if(s)
-        return (s->getPerimeter() > 20);
-    return false;
-}
+auto perimeterBiggerThan20 = [](shared_ptr<Shape> s) {if(s){
+        return (s->getPerimeter() > 20); } return false; };
 
-bool areaLessThan10(shared_ptr<Shape> s)
-{
-    if(s)
-        return (s->getArea() < 10);
-    return false;
-}
+auto areaLessThanX = [x = 10](shared_ptr<Shape> s) {if(s){
+    return (s->getArea() < x);} 
+    return false; };
 
-void printCollection(const Collection& collection)
-{
-    for (const auto & it : collection)
+void printCollection(const Collection& collection) {
+    for (const auto& it : collection)
         if (it)
             it->print();
 }
 
-void printAreas(const Collection& collection)
-{
-    for (const auto & it : collection)
+void printAreas(const Collection& collection) {
+    for (const auto& it : collection)
```

Comment:
I think you should fix formatting by adding bracets to oneliners,since it will make code more readable.